federation/pkg/kubefed/init/init.go

```diff
@@ -125,6 +125,10 @@ var (
 		"app":    "federated-cluster",
 		"module": "federation-controller-manager",
 	}
+
+	validDNSProviders = []string{
```

Comment:
Have you tried using RegisteredDnsProviders?
Example usage: https://github.com/kubernetes/kubernetes/blob/5af81b0955d907256a222e5cbcccfcba0db88b53/federation/cmd/federation-controller-manager/app/options/options.go#L126